objects/src/lib.rs

```diff
@@ -1,6 +1,10 @@
 #![cfg_attr(not(feature = "std"), no_std)]
 
-#[cfg(not(feature = "std"))]
+#[cfg(all(not(feature = "std"), not(test)))]
+extern crate alloc;
+
+#[cfg(all(not(feature = "std"), test))]
```

Comment:
This looks different to what I'd expect.  Normally we'd be able to run just specify `no_std`.  Why is it needed?

Reply:
the `vec!` macro is only used in the tests atm, without this `cargo check --no-default-features --all-targets` fails:

```
error: cannot find macro `vec` in this scope
  --> objects/src/notes/inputs.rs:66:29
   |
66 |     let expected_ordering = vec![
   |                             ^^^
   |
   = help: consider importing this macro:
           alloc::vec

error: cannot find macro `vec` in this scope
  --> objects/src/notes/inputs.rs:64:18
   |
64 |     let inputs = vec![Felt::new(1), Felt::new(2), Felt::new(3)];
   |                  ^^^
   |
   = help: consider importing this macro:
           alloc::vec

error: could not compile `miden-objects` due to 2 previous errors
```

Reply:
That makes sense.  I'm not a fan of the additional complexity in the feature flags.  My preference would be to change the tests to use `Vec::from([..])`

Reply:
changed

Reply:
I think this is fine. The other alternative was to import `vec` from `crypto::utils::collections`.

Reply:
@bobbinth I think the problem was that we needed `#[macro_use]` for `alloc` to use the `vec![]` macro.  However we only use `vec![]` in tests so clippy would have complained.  Once we use the `vec![]` macro in the library we can introduce `#[macro_use]` without clippy complaining.

Reply:
Ah! Makes sense.